date_range/static/src/js/date_range.js

```diff
@@ -0,0 +1,125 @@
+/* Copyright 2016 ACSONE SA/NV (<http://acsone.eu>)
+ * License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl). */
+odoo.define('date_range.search_filters', function (require) {
+"use strict";
+
+var core = require('web.core');
+var data = require('web.data');
+var filters = require('web.search_filters');
+var rpc = require('web.rpc');
+var framework = require('web.framework');
+
+var _t = core._t;
+filters.ExtendedSearchProposition.include({
+    select_field: function (field) {
+        this._super.apply(this, arguments);
+        this.is_date_range_selected = false;
+        this.is_date = field.type === 'date' || field.type === 'datetime';
+        this.$value = this.$el.find('.searchview_extended_prop_value, .o_searchview_extended_prop_value');
+        if (this.is_date){
+            var ds = new data.DataSetSearch(this, 'date.range.type', this.context, [[1, '=', 1]]);
+            ds.read_slice(['name'], {}).done(this.proxy('add_date_range_types_operator'));
+        }
+    },
+
+    add_date_range_types_operator: function (date_range_types){
+        var self = this;
+        _.each(date_range_types, function (drt) {
+            $('<option>', {value: 'drt_' + drt.id})
+                .text(_('in ') + drt.name)
+                .appendTo(self.$el.find('.searchview_extended_prop_op, .o_searchview_extended_prop_op'));
+        });
+    },
+
+    operator_changed: function (e) {
+        var val = $(e.target).val();
+        this.is_date_range_selected = val.startsWith('drt_');
+        if (this.is_date_range_selected){
+            var type_id = val.replace('drt_', '');
+            this.date_range_type_operator_selected(type_id);
+            return;
+        }
+        this._super.apply(this, arguments);
+    },
+
+    date_range_type_operator_selected: function (type_id){
+        this.$value.empty().show();
+        var ds = new data.DataSetSearch(this, 'date.range', this.context, [['type_id', '=', parseInt(type_id)]]);
+        ds.read_slice(['name','date_start', 'date_end'], {}).done(this.proxy('on_range_type_selected'));
+
+    },
+
+    on_range_type_selected: function (date_range_values){
+        this.value = new filters.ExtendedSearchProposition.DateRange(this, this.value.field, date_range_values);
+        this.value.appendTo(this.$value);
+        if (!this.$el.hasClass('o_filter_condition')){
+            this.$value.find('.date-range-select').addClass('form-control');
+        }
+        this.value.on_range_selected();
+    },
+
+    get_filter: function () {
+        var res = this._super.apply(this, arguments);
+        if (this.is_date_range_selected){
+            // in case of date.range, the domain is provided by the server and we don't
+            // want to put nest the returned value into an array.
+            res.attrs.domain = this.value.domain;
+        }
+        return res;
+    }
+
+});
+
+/**
+Since Odoo 11, The Field class used as base class for all specialized filter
```

Comment:
Is this still the case in Odoo 11? If so can you change it to Odoo 12

Reply:
I would keep it as it is, meaning that:
_starting from V11, the Field class used as base class for all specialized filter etc..._